cloudinit/sources/DataSourceOVF.py

```diff
@@ -389,20 +399,31 @@ def get_network_config(nics=None, nameservers=None, search=None):
 
 # This will return a dict with some content
 #  meta-data, user-data, some config
-def read_vmware_imc(config):
+def read_vmware_imc(config, paths=None):
     md = {}
     cfg = {}
     ud = None
-    if config.host_name:
-        if config.domain_name:
-            md['local-hostname'] = config.host_name + "." + config.domain_name
-        else:
-            md['local-hostname'] = config.host_name
+    if config is not None:
+        if config.host_name:
+            if config.domain_name:
+                md['local-hostname'] = config.host_name + "." +\
+                                       config.domain_name
+            else:
+                md['local-hostname'] = config.host_name
+        if config.timezone:
+            cfg['timezone'] = config.timezone
 
-    if config.timezone:
-        cfg['timezone'] = config.timezone
+        md['instance-id'] = "iid-vmware-imc"
+    else:
+        # Set instance-id to be same with current instance-id if it exists
+        # when there is no config
+        dp = paths.get_cpath('data')
+        iid_fn = os.path.join(dp, 'instance-id')
+        try:
+            md['instance-id'] = util.load_file(iid_fn).strip()
+        except Exception:
+            md['instance-id'] = "iid-vmware-imc"
```

Comment:
I think you should implement check_instance_id() for OVF instead of this; which is somewhat equivalent.  However, you should do some verification that the instance-id value itself is what you expect (one of the various iid-dsovf, iid-vmware-imc) values, otherwise something has booted this image somewhere else than on VMWare and you should not return True for _get_data().

Reply:
Set the instance-id to be same with existing instance-id only when self. _vmware_cust_enabled is true which means it's on VMware platform, and only when there is no configuration after waiting which means no VMware customization happening, so keep whatever instance-id as it is.